### ISSUE TITLE
Integration not starting up

### ISSUE THREAD

**Issue by dagwud** [swartjean/ha-eskom-loadshedding#17]
## Version of the custom_component
1.0.6 (according to const.py)

## Configuration
No specific config - installed months ago via HACS and I don’t recall configuring anything specific. If there are config details needed please point me to them?

## Describe the bug
Pretty sure this would be caused by something on Eskom’s side, but would it be possible to fail more gracefully?

Integration doesn’t seem to be starting up.

The entity shows up as “not available” under Devices & Services > Eskom Loadshedding Interface > 1 entity
![BC481059-386A-4BF4-AA39-EEDC36706F07](https://user-images.githubusercontent.com/5947216/152220295-5ad10364-d7f7-4e35-a38b-da1171c1858a.jpeg)

## Debug log

```text

2021-02-02 20:43:40 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: eskom_loadshedding

```
That’s all I can seem to find (apart from all my templates in dashboards breaking because of the missing entity) with debug logs on. If there’s some specific way of increasing the log level beyond debug for just this component, happy to try that)

**Comment by dagwud** [comment]
More info:

I redownloaded the plug-in via HACS - no change, it still doesn’t start up

**Comment by dagwud** [comment]
After completely removing the integration, restarting HA, reinstalling the repo, restarting again, resetting my front end cache, and then configuring the integration, it’s now working.

**Comment by swartjean** [comment]
Hi @dagwud, thanks for raising this. I have also noticed some issues with the integration starting up when Eskom's API is being unreliable (i.e. when the requests to https://loadshedding.eskom.co.za/LoadShedding/GetStatus time out). Their API is pretty rough around the edges at the best of times.

I checked this endpoint a few times yesterday when I got the email for your issue and noticed that it wasn't responding.

I'll see if I can get it to fail more gracefully, maybe by adding an "Unknown" state. If it looks promising I'll reopen this issue and act on it with a new release.
